core/cache/model.go

```diff
@@ -315,6 +338,40 @@ func (m *Model) removeMachine(ch RemoveMachine) error {
 	return nil
 }
 
+// updateBranch adds or updates the branch in the model.
+// Only "in-flight" branches should ever reside in the change.
+// A committed or aborted branch (with a non-zero time-stamp for completion)
+// should be passed through by the cache worker as a deletion.
+func (m *Model) updateBranch(ch BranchChange, rm *residentManager) {
+	m.mu.Lock()
```

Comment:
Why not do `defer m.doLocked()()` for consistency with the methods surrounding this one?

Reply:
It is actually consistent with the other `update<Entity>` methods. That explicitly lock/unlock (they do not have multiple points of return).